tracer/src/Datadog.Trace/Sampling/CustomSamplingRule.cs

```diff
@@ -115,18 +127,16 @@ private static string WrapWithLineCharacters(string regex)
             return regex;
         }
 
-        private bool DoesNotMatch(string input, string pattern)
+        private bool DoesNotMatch(string input, Regex regex)
         {
             try
             {
-                if (pattern != null &&
-                    !Regex.IsMatch(
-                        input: input,
-                        pattern: pattern,
-                        options: RegexOptions.None,
-                        matchTimeout: RegexTimeout))
+                if (regex is not null)
                 {
-                    return true;
+                    if (!regex.Match(input).Success)
```

Comment:
nit-picking: looks like you can combine this two `if` conditions:
```csharp
if(regex is not null && !regex.Match(input).Success)
```
...or replace the `if` with
```csharp
return regex is not null && !regex.Match(input).Success
```
and return `false` in the `catch` clause

Reply:
bonus nit-pick: I find reverse logic like `DoesNotMatch()` confusing. I would not mind if you invert the logic to be
```csharp
bool Matches(...)
{
   return regex is not null && regex.Match(input).Success;
}
```
at which point it looks like you can just inline this method into `IsMatch()` above. to make things simpler.

Reply:
I like it, updated in fce030aa4c98b513236d5f9e43269a40abd9fb10